eventuate-log-cassandra/src/main/scala/com/rbmhtechnology/eventuate/log/cassandra/CassandraEventLog.scala

```diff
@@ -183,25 +184,28 @@ class CassandraEventLog(id: String, aggregateIndexing: Boolean) extends EventLog
     eventLogStore.write(events, partition)
     updateCount += events.size
     if (updateCount >= cassandra.settings.indexUpdateLimit) {
-      if (aggregateIndexing) {
-        // asynchronously update the index
-        index ! UpdateIndex(null, clock.sequenceNr)
-      } else {
-        // otherwise update the event log clock snapshot only
-        import context.dispatcher
-        indexStore.writeEventLogClockSnapshotAsync(clock)
-      }
-
+      updateIndex(clock)
       updateCount = 0L
     }
   }
 
+  private def updateIndex(clock: EventLogClock): Future[EventLogClock] = {
+    import context.dispatcher
+    if (aggregateIndexing) {
+      // asynchronously update the index
+      val promise = Promise[UpdateIndexSuccess]
+      index ! UpdateIndex(null, clock.sequenceNr, promise)
+      promise.future.pipeTo(self)
+      promise.future.map(_.clock)
+    } else
+      // otherwise update the event log clock snapshot only
+      indexStore.writeEventLogClockSnapshotAsync(clock)
+  }
+
   override def unhandled(message: Any): Unit = message match {
     case u @ UpdateIndexSuccess(clock, _) =>
       indexSequenceNr = clock.sequenceNr
       onIndexEvent(u)
-    case u @ UpdateIndexFailure(_) =>
```

Comment:
It actually never was required as this is anyways only used in tests for inspecting the index progress and since there was no test inspecting failures I thought its ok to remove it here.